iroh-net/src/endpoint.rs

```diff
@@ -432,10 +432,12 @@ impl Endpoint {
 
     /// Connects to a remote [`Endpoint`].
     ///
-    /// A [`NodeAddr`] is required. It must contain the [`NodeId`] to dial and may also
-    /// contain a [`RelayUrl`] and direct addresses. If direct addresses are provided, they
-    /// will be used to try and establish a direct connection without involving a relay
-    /// server.
+    /// A value that can be converted into a [`NodeAddr`] is required. This can be either a
+    /// [`NodeAddr`], a [`NodeId`] or a [`iroh_base::ticket::NodeTicket`].
+    ///
+    /// The [`NodeAddr`] must contain the [`NodeId`] to dial and may also contain a [`RelayUrl`]
```

Comment:
We don't use `dial` in any of the API names of `iroh-net::Endpoint` so I prefer not to use that in documentation either as that is probably confusing to users.  I usually stick to "connect", "establish a connection" or something like that.

(yeah, there's the somewhat weird dialer module which is a bit of an odd duck...)